Lib/test/test_subprocess.py

```diff
@@ -1485,6 +1485,50 @@ def test_exception_bad_args_0(self):
         else:
             self.fail("Expected OSError: %s" % desired_exception)
 
+    # We mock the __del__ method for Popen in the next two tests
+    # because it does cleanup based on the pid returned by fork_exec
+    # along with issuing a resource warning if it still exists. Since
+    # we don't actually spawn a process in these tests we can forego
+    # the destructor. An alternative would be to set _child_created to
+    # False before the destructor is called but there is no easy way
+    # to do that
+    @mock.patch.object(subprocess.Popen, "__del__")
```

Comment:
You can never control when a `__del__` destructor is going to be called. So this mock may well have been undone before it does get called by the gc.  Never mock out anything on a class that is called outside of the test's control flow such as ` __del__`.

If you want to replace `__del__` behavior something different due to the way a test works, one way to do that is using a subclass that overrides it and does not call the parent class's method.